components/director/internal/domain/document/resolver_test.go

```diff
@@ -139,11 +211,13 @@ func TestResolver_AddDocument(t *testing.T) {
 
 	for _, testCase := range testCases {
 		t.Run(testCase.Name, func(t *testing.T) {
+			persistTx := testCase.PersistenceFn()
+			transact := testCase.TransactionerFn(persistTx)
 			svc := testCase.ServiceFn()
 			appSvc := testCase.AppServiceFn()
 			converter := testCase.ConverterFn()
 
-			resolver := document.NewResolver(svc, appSvc, nil)
+			resolver := document.NewResolver(transact, svc, appSvc, nil)
```

Comment:
Add `transact.AssertExpectations(t)` and `persistTx.AssertExpctations` at the end of the test